cee_syslog_handler/__init__.py

```diff
@@ -271,3 +271,50 @@ def filter(self, record):
         """
         found = self._pattern.search(record.getMessage())
         return not found
+
+
+class RegexRedactFilter(logging.Filter):
+    """
+    This filter redacts parts of log messages and exception traces
+    by substituting through a provided regular expression.
+    Use with caution: with great power comes great responsibility
+    """
+    def __init__(self, filter_regex=None,
+                 replace_string="<redacted>"):
+        super(RegexRedactFilter, self).__init__()
+        self._pattern = re.compile(filter_regex)
+        self._replacement = replace_string
+
+    def redact(self, string):
+        return re.sub(self._pattern, self._replacement, string)
+
+    def filter(self, record):
+        if not self._pattern.search(record.getMessage())\
+                and not record.exc_info:
+            return True
+
+        message = record.getMessage()
+        record.msg = self.redact(message)
+        record.args = ()
+
+        # use methods from formatter to safely redact exception
+        # and stack traces
+        formatter = logging.Formatter()
+
+        if record.exc_info:
```

Comment:
I am wondering if we can eliminate writing this on our own if we use `logging.Formatter.format(record)` here. It seems to take care of fusing the formatted exception with the log message.

Reply:
It does fuse the formatted exception with the log message, but it also applies a supplied style containing things like loglevel, name and other log message formatting details.
That would lead to log messages like:
`%(loglevel):%(name):%(loglevel:name:message)`